### PR TITLE
Rework reified markers format

### PR DESCRIPTION
**Pull request by dzharkov**

### CODE REVIEW

File: compiler/backend/src/org/jetbrains/kotlin/codegen/optimization/common/Util.kt [hunk 1]
```diff
@@ -97,3 +96,24 @@ inline fun AbstractInsnNode.findPreviousOrNull(predicate: (AbstractInsnNode) ->
...
+//           mv.visitIntInsn(Opcodes.SIPUSH, cst);
+//       } else {
+//           mv.visitLdcInsn(new Integer(cst));
+//       }
+//   }
+val AbstractInsnNode.intConstant: Int? get() =
+    when (opcode) {
+        in ICONST_0..ICONST_5 -> opcode - ICONST_0
+        BIPUSH, SIPUSH -> (this as IntInsnNode).operand
+        LDC -> (this as LdcInsnNode).cst as? Int?
```
Comment (udalov) [inline compiler/backend/src/org/jetbrains/kotlin/codegen/optimization/common/Util.kt]: `as? Int` should work I think

File: compiler/backend/src/org/jetbrains/kotlin/codegen/optimization/common/Util.kt [hunk 2]
```diff
@@ -97,3 +96,24 @@ inline fun AbstractInsnNode.findPreviousOrNull(predicate: (AbstractInsnNode) ->
...
+//           mv.visitIntInsn(Opcodes.BIPUSH, cst);
+//       } else if (cst >= Short.MIN_VALUE && cst <= Short.MAX_VALUE) {
+//           mv.visitIntInsn(Opcodes.SIPUSH, cst);
+//       } else {
+//           mv.visitLdcInsn(new Integer(cst));
+//       }
+//   }
+val AbstractInsnNode.intConstant: Int? get() =
+    when (opcode) {
+        in ICONST_0..ICONST_5 -> opcode - ICONST_0
```
Comment (udalov) [inline compiler/backend/src/org/jetbrains/kotlin/codegen/optimization/common/Util.kt]: What about `ICONST_M1`?

File: compiler/backend/src/org/jetbrains/kotlin/codegen/codegenUtil.kt [hunk 3]
```diff
@@ -56,11 +60,37 @@ fun generateIsCheck(
         }
     }
     else {
-        generateInstanceOfInstruction(v)
+        TypeIntrinsics.instanceOf(v, kotlinType, asmType)
+    }
+}
+
+fun generateAsCheck(
```
Comment (udalov) [inline compiler/backend/src/org/jetbrains/kotlin/codegen/codegenUtil.kt]: `is` is a "check", but `as` is not :) `generateCast`?

**Comment by dzharkov** [comment]
Ok, thanks.
I'll squash my fixes and merge manually

**Pull request closed without merging**